activerecord/test/cases/calculations_test.rb

```diff
@@ -460,6 +460,44 @@ def test_should_count_manual_select_with_include
     assert_equal 6, Account.select("DISTINCT accounts.id").includes(:firm).count
   end
 
+  def test_should_count_manual_select_with_count_all
+    assert_equal 5, Account.select("DISTINCT accounts.firm_id").count(:all)
+  end
+
+  def test_should_count_with_manual_distinct_select_and_distinct
+    assert_equal 4, Account.select("DISTINCT accounts.firm_id").distinct(:id).count
+  end
+
+  def test_should_count_manual_select_with_group_with_count_all
+    c = Account.select("DISTINCT accounts.firm_id").group("accounts.firm_id").count(:all)
+    assert_equal([[nil, 1], [1, 1], [2, 1], [6, 2], [9, 1]].to_h, c)
+  end
+
+  def test_should_count_manual_with_count_all
+    assert_equal 6, Account.count(:all)
+  end
+
+  def test_size_manual_select_distinct
+    assert_equal 5, Account.select("DISTINCT accounts.firm_id").size
+  end
+
+  def test_size_with_manual_distinct_select_with_loaded_size
+    size = Account.select("DISTINCT accounts.firm_id").size
+    loaded_size = Account.select("DISTINCT accounts.firm_id").load.size
+    assert_equal size, loaded_size
+  end
+
+  def test_size_with_distinct_with_loaded_size
+    size = Account.distinct(:firm_id).size
+    loaded_size = Account.distinct(:firm_id).load.size
+    assert_equal size, loaded_size
+  end
+
+  def test_size_with_manual_select_with_group_with
+    c = Account.select("DISTINCT accounts.firm_id").group("accounts.firm_id").size
```

Comment:
This should return 5, since it should be the same result with `Account.select("DISTINCT accounts.firm_id").group("accounts.firm_id").load.size`.

Reply:
Somehow executing:
```ruby
Account.select("DISTINCT accounts.firm_id").group("accounts.firm_id").size
# { nil => 1, 1 => 1, 2 => 1, 6 => 2, 9 => 1 }
```
Now returns the hash. But I agree that it should return the same of: 
```ruby
Account.select("DISTINCT accounts.firm_id").group("accounts.firm_id").load.size
# 5
```
Should I investigate this for this PR ?

Reply:
Feel free to create another PR.

Reply:
As an user, I expect `.group.size` (same as `.group.count`) to return an hash. And doing `.load.size` / `to_a.size` would return a collection of the instances of the model and then how many of them there are.

Reply:
Originally, `relation.size` behaves fragile, that is a breaking change though.

```ruby
relation = Post.left_joins(:comments).group(:id)
relation.size # => return a hash
relation.each { |record| puts record.id }
relation.size # => return an integer

relation = Post.left_joins(:comments).group(:id)
relation.count # => return a hash
relation.each { |record| puts record.id }
relation.count # => return a hash

relation = Post.left_joins(:comments).group(:id)
relation.length # => return an integer
relation.each { |record| puts record.id }
relation.length # => return an integer
```

Reply:
But in my opinion `size` and `load.size` should have the same return value the difference should be that when using just `size` if the collection is loaded it will not execute a query but if it isn't it executes a query without loading the collection instances. 

That they return different values when the collection is loaded and when it isn't is confusing.